src/mpid/ch4/src/ch4r_symheap.h

```diff
@@ -450,38 +493,31 @@ static inline int MPIDI_CH4R_get_shm_symheap(MPI_Aint shm_size, MPI_Aint * shm_o
                                          MAP_PRIVATE | MAP_ANON | MAP_FIXED, -1, 0, MPL_MEM_RMA);
                 } else
                     *base_ptr = (void *) MAP_FAILED;
-                mapfail_flag = (*base_ptr == (void *) MAP_FAILED) ? 1 : 0;
+                map_result = (*base_ptr == (void *) MAP_FAILED) ?
+                    MPIDI_CH4I_SYMSHM_MAP_FAIL : MPIDI_CH4I_SYMSHM_SUCCESS;
             }
         }
 
         /* check if any mapping failure occurs */
-        mpi_errno = MPIR_Allreduce(&mapfail_flag, &any_mapfail_flag, 1, MPI_UNSIGNED,
-                                   MPI_BOR, comm, &errflag);
+        mpi_errno = MPIR_Allreduce(&map_result, &all_map_result, 1, MPI_INT,
+                                   MPI_MAX, comm, &errflag);
         MPIR_ERR_CHKANDJUMP(errflag, mpi_errno, MPI_ERR_OTHER, "**coll_fail");
 
         /* cleanup local shm segment if mapping failed */
-        if (any_mapfail_flag && mapsize > 0) {
-            if (shm_comm_ptr != NULL) {
-                if (mapfail_flag) {
-                    /* if locally map failed, only release handle */
-                    int mpl_err = MPL_SHM_SUCCESS;
-                    mpl_err = MPL_shm_hnd_finalize(shm_segment_hdl_ptr);
-                    MPIR_ERR_CHKANDJUMP(mpl_err, mpi_errno, MPI_ERR_OTHER, "**remove_shar_mem");
-                } else {
-                    /* destroy successful shm segment */
-                    mpi_errno = MPIDI_CH4U_destroy_shm_segment(mapsize, shm_segment_hdl_ptr,
-                                                               base_ptr);
-                    if (mpi_errno)
-                        MPIR_ERR_POP(mpi_errno);
-                }
-            } else if (!mapfail_flag) {
+        if (all_map_result != MPIDI_CH4I_SYMSHM_SUCCESS && mapsize > 0) {
+            if (shm_comm_ptr != NULL && map_result == MPIDI_CH4I_SYMSHM_SUCCESS) {
+                /* destroy successful shm segment */
+                mpi_errno = MPIDI_CH4U_destroy_shm_segment(mapsize, shm_segment_hdl_ptr, base_ptr);
+                if (mpi_errno)
+                    MPIR_ERR_POP(mpi_errno);
+            } else if (map_result == MPIDI_CH4I_SYMSHM_SUCCESS) {
```

Comment:
Can this block be:
```C
if (all_map_result != MPIDI_CH4I_SYMSHM_SUCCESS && mapsize > 0 && map_result == MPIDI_CH4I_SYMSHM_SUCCESS) {
     if (shm_comm_ptr != NULL) {
     } else {
     }
}
```
(no logic change, just a style preference.)

Reply:
It makes sense. fixed.